src/packages/@ncigdc/modern_components/GeneExpression/inchlib/index.js

```diff
@@ -3310,6 +3250,84 @@ import ageDisplay, { getLowerAgeYears } from '@ncigdc/utils/ageDisplay';
       link.click();
       document.body.removeChild(link);
     }
+
+    return new Promise((resolve, reject) => {
+      const width = self.stage.width();
+      const height = self.stage.height();
+      const img_file_name = `gene-expression-values.${
+        moment().format('YYYY-MM-DD-HHmmss')
+      }.${img_format}`;
+
+      $.multitask.add(() => {
```

Comment:
Here, we put the resource-intensive part of the image generation in a time buffered queue, so it allows for other code to be executed meanwhile.